### PR TITLE
Throw better exception when failing to load plugin

### LINKED ISSUE #446: Poor error message for package.json exports pointing to non-existent path
When the file specified in a `package.json`'s `"main"` entry doesn't exist, `eslint-doc-generator` exits with an only somewhat informative message:

```plaintext
$ npm run update:eslint-docs

> eslint-plugin-expect-type@0.2.2 update:eslint-docs
> eslint-doc-generator

Unable to determine plugin entry point.
```

Is there a way this package could provide a more clear error message? For example:

```plaintext
Plugin entry point `dist/index.js` does not appear to exist on disk.
```

See https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/tree/7c55e1e6cda11eeeb4a7fdefd6e322cb32154d17:

```shell
git clone https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type
cd eslint-plugin-expect-type
git checkout 7c55e1e6cda11eeeb4a7fdefd6e322cb32154d17
npm i
npm run update:eslint-docs
```

The issue is that the package specifies `"main": "./dist/index.js"` and needs to run `npm run build` to generate that before `npm run update:eslint-docs`.

[Example GitHub Actions run](https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/actions/runs/4885810867/jobs/8720352650)

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #446.

Please forgive me for rushing ahead, I was excited about the bugfix. 😄

### CODE REVIEW

File: test/lib/generate/cjs-test.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ describe('generate (cjs)', function () {
         'cjs-main-file-does-not-exist'
       );
       await expect(generate(FIXTURE_PATH)).rejects.toThrow(
-        'Unable to determine plugin entry point.'
```
Comment (JoshuaKGoldberg) [inline test/lib/generate/cjs-test.ts:71]: I tried to construct a `package.json` test case to throw the existing default error... and couldn't figure it out.

**Comment by JoshuaKGoldberg** [comment]
@bmish sorry for dropping the ball on this one - totally slipped off my task list. Looks like `CI / build` is an unrelated out-of-memory issue. But this should be good to review!

File: test/lib/generate/cjs-test.ts (hunk 1 above)
Comment (bmish) [reply test/lib/generate/cjs-test.ts:71]: I'll try to take a look at this so we can test the old exception still.

**Comment by bmish** [comment]
@JoshuaKGoldberg no problem and sorry for the delayed review. I will see if I can add a test and get CI cleaned up (#288) in the near future so we can merge this.

**Review by bmish — APPROVED** [review]
I'm not sure how to improve the tests further, so I'm going to go ahead and merge this. Sorry again about the delay.

**Review by bmish — APPROVED** [review]
I'm not sure how to improve the tests further, so I'm going to go ahead and merge this. Sorry again about the delay.

**Comment by JoshuaKGoldberg** [comment]
Woohoo! I'm just happy it's in, thanks!

**Pull request merged**